lib/src/fretting.dart

```diff
@@ -126,42 +142,62 @@ class FretPosition {
   final int fretNumber;
   final int semitones;
 
-  FretPosition({int this.stringIndex, int this.fretNumber, int this.semitones});
-  bool operator ==(FretPosition other) => stringIndex == other.stringIndex && fretNumber == other.fretNumber;
+  FretPosition({this.stringIndex, this.fretNumber, this.semitones});
+  // bool operator ==(FretPosition other) =>
+  //     stringIndex == other.stringIndex && fretNumber == other.fretNumber;
+  @override
+  bool operator ==(dynamic other) {
+    final FretPosition typedOther = other;
+    return stringIndex == typedOther.stringIndex &&
+        fretNumber == typedOther.fretNumber;
+  }
+
   int get hashCode => 37 * stringIndex + fretNumber;
   String toString() => "$stringIndex.$fretNumber($semitones)";
-  String get inspect => {'string': stringIndex, 'fret': fretNumber, 'semitones': semitones}.toString();
+  String get inspect => {
+        'string': stringIndex,
+        'fret': fretNumber,
+        'semitones': semitones
+      }.toString();
 }
 
-Set<FretPosition> chordFrets(Chord chord, FrettedInstrument instrument, int highestFret) {
+Set<FretPosition> chordFrets(
+    Chord chord, FrettedInstrumentClass instrument, int highestFret) {
   var positions = new Set<FretPosition>();
   var semitoneSet = chord.pitches.map((pitch) => pitch.semitones % 12).toSet();
   eachWithIndex(instrument.stringPitches, (Pitch pitch, int stringIndex) {
     for (var fretNumber = 0; fretNumber <= highestFret; fretNumber++) {
-      var semitones = instrument.pitchAt(stringIndex: stringIndex, fretNumber: fretNumber).semitones;
+      var semitones = instrument
+          .pitchAt(stringIndex: stringIndex, fretNumber: fretNumber)
+          .semitones;
       if (semitoneSet.contains(semitones % 12)) {
-        var position = new FretPosition(stringIndex: stringIndex, fretNumber: fretNumber, semitones: semitones);
+        var position = new FretPosition(
+            stringIndex: stringIndex,
+            fretNumber: fretNumber,
+            semitones: semitones);
         positions.add(position);
       }
     }
   });
   return positions;
 }
 
-List<Fretting> chordFrettings(Chord chord, FrettedInstrument instrument, {int highestFret: 4}) {
+List<Fretting> chordFrettings(Chord chord, FrettedInstrumentClass instrument,
+    {highestFret: 4}) {
   var warn = false;
   int minPitchClasses = chord.intervals.length;
+  print(warn);
```

Comment:
Delete `print(warn);`?

Reply:
The only reason I added the `print(warn);` was because it wasn’t being used and I was getting rid of the dart warning.